lib/utils/custom_disissible.dart

```diff
@@ -0,0 +1,165 @@
+import 'package:flutter/material.dart';
+
+class CustomDismissible extends StatefulWidget {
+  const CustomDismissible({
+    super.key,
+    required this.child,
+    this.onDismissed,
+    this.dismissThreshold = 0.2,
+    this.enabled = true,
+    this.handleDragStart,
+    this.handleDragEnd,
+  });
+
+  final Widget child;
+  final double dismissThreshold;
+  final VoidCallback? onDismissed;
+  final bool enabled;
+  final Function(DragStartDetails dragStartDetails)? handleDragStart;
+  final Function(DragEndDetails dragEndDetails)? handleDragEnd;
+
+  @override
+  State<CustomDismissible> createState() => _CustomDismissibleState();
+}
+
+class _CustomDismissibleState extends State<CustomDismissible>
+    with SingleTickerProviderStateMixin {
+  late AnimationController _animateController;
+  late Animation<Offset> _moveAnimation;
+  late Animation<double> _scaleAnimation;
+  late Animation<Decoration> _opacityAnimation;
+
+  double _dragExtent = 0;
+  bool _dragUnderway = false;
+
+  bool get _isActive => _dragUnderway || _animateController.isAnimating;
+
+  @override
+  void initState() {
+    super.initState();
+
+    _animateController = AnimationController(
+      duration: const Duration(milliseconds: 300),
+      vsync: this,
+    );
+
+    _updateMoveAnimation();
+  }
+
+  @override
+  void dispose() {
+    _animateController.dispose();
+
+    super.dispose();
+  }
+
+  void _updateMoveAnimation() {
+    final double end = _dragExtent.sign;
+
+    _moveAnimation = _animateController.drive(
+      Tween<Offset>(
+        begin: Offset.zero,
+        end: Offset(0, end),
+      ),
+    );
+
+    _scaleAnimation = _animateController.drive(
+      Tween<double>(
+        begin: 1,
+        end: 0.5,
+      ),
+    );
+
+    _opacityAnimation = DecorationTween(
+      begin: const BoxDecoration(
+        color: Color(0xFF000000),
+      ),
+      end: const BoxDecoration(
+        color: Color(0x00000000),
+      ),
+    ).animate(_animateController);
```

Comment:
Why don't you use color M3

Reply:
Don't have design for it yet, but anyway, I change it a little bit to easy to understand